misc.lua

```diff
@@ -51,4 +54,39 @@ function flib_misc.ticks_to_timestring(tick)
   end
 end
 
+function flib_misc.get_signal_value(entity, signal)
+  local red = entity.get_circuit_network(wire_type.red)
+  local green = entity.get_circuit_network(wire_type.green)
+  local value = 0
+  if red then
+    value = red.get_signal_value(signal)
+  end
+  if green then
+    value = value + green.get_signal_value(signal)
+  end
+  return value
+end
+
+function flib_misc.check_signal(signal, blacklist_table)
+  if type(signal) == "table" then
+    local name = signal.name
+    local signaltype = signal.type
+    if blacklist_table[name] or (signaltype == "fluid" and not game.fluid_prototypes[name]) or (signaltype == "item" and not game.item_prototypes[name]) or signaltype == "virtual" and not game.virtual_signal_prototypes[name] then
```

Comment:
use `if (blacklist_table and blacklist_table[name]) or ...` to make blacklist optional.

Reply:
Yea, forgot that sry.